core/src/main/scala/org/http4s/otel4s/OtelMetrics.scala

```diff
@@ -0,0 +1,273 @@
+ *
+ * {prefix}.response.duration{labels=classifier,method,phase} - Histogram
+ *
+ * {prefix}.active_request.count{labels=classifier} - Gauge
+ *
+ * {prefix}.request.count{labels=classifier,method,status} - Counter
+ *
+ * {prefix}.abnormal_terminations{labels=classifier,termination_type} - Histogram
+ *
+ * Labels --
+ *
+ * method: Enumeration values: get, put, post, head, move, options, trace, connect, delete, other
+ *
+ * phase: Enumeration values: headers, body
+ *
+ * code: Enumeration values: 1xx, 2xx, 3xx, 4xx, 5xx
+ *
+ * termination_type: Enumeration values: abnormal, error, timeout
+ */
+object OtelMetrics {
+
+  /** Creates a [[MetricsOps]] that supports OpenTelemetry metrics
+   *
+   * @param prefix
+   * a prefix that will be added to all metrics
+   */
+  def metricsOps[F[_] : Sync : Meter](
+                                       prefix: String = "org.http4s.server",
+                                       responseDurationSecondsHistogramBuckets: BucketBoundaries = DefaultHistogramBuckets,
+                                     ): F[MetricsOps[F]] =
+    for {
+      metrics <- createMetricsCollection(prefix, responseDurationSecondsHistogramBuckets)
+    } yield createMetricsOps(metrics)
+
+  private def createMetricsOps[F[_] : Sync](metrics: MetricsCollection[F]): MetricsOps[F] =
+    new MetricsOps[F] {
+      override def increaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .inc(Attribute("classifier", label(classifier)))
+
+      override def decreaseActiveRequests(classifier: Option[String]): F[Unit] =
+        metrics.activeRequests
+          .dec(Attribute("classifier", label(classifier)))
+
+      override def recordHeadersTime(
+                                      method: Method,
+                                      elapsed: Long,
+                                      classifier: Option[String]
+                                    ): F[Unit] =
+        metrics.responseDuration
+          .record(
+            secondsFromNanos(elapsed),
+            Attribute("classifier", label(classifier)),
+            Attribute("method", reportMethod(method)),
+            Attribute("phase", Phase.report(Phase.Headers))
+          )
+
+      override def recordTotalTime(
+                                    method: Method,
+                                    status: Status,
+                                    elapsed: Long,
+                                    classifier: Option[String]
+                                  ): F[Unit] =
+        metrics.responseDuration
+          .record(
+            secondsFromNanos(elapsed),
+            Attribute("classifier", label(classifier)),
+            Attribute("method", reportMethod(method)),
+            Attribute("phase", Phase.report(Phase.Body))
+          ) *>
+          metrics.requests
+            .inc(
+              Attribute("classifier", label(classifier)),
+              Attribute("method", reportMethod(method)),
```

Comment:
OpenTelemetry has a standardized naming for some attributes.

For example: https://github.com/typelevel/otel4s/blob/main/semconv/stable/src/main/scala/org/typelevel/otel4s/semconv/HttpAttributes.scala#L70

Attributes in the `otel4s-semconv` package are generated from the open telemetry **stable** semantic conventions.

Reply:
Doesn't seem like it. I left the otel4s-semconv dep in for now.